google/oauth2/_client_async.py

```diff
@@ -83,26 +83,44 @@ async def _token_endpoint_request_no_throw(
             else response_body1
         )
 
-        response_data = json.loads(response_body)
+        try:
+            response_data = json.loads(response_body)
+        except ValueError:
+            response_data = response_body
+
+        return response, response_data
+
+    response, response_data = await _perform_request()
```

Comment:
similarly, should be a way to refactor so we do perform_request and check retryable_error one